Add and fix linting

hugovk opened the pull request:
* Fixes #12 
* Includes and closes #35
* Switch from `pylint` to `flake8` linting, which includes `pyflakes` (checks for errors) and `pycodestyle` (formerly called `pep8`)
* Fix linting errors, including running `isort`

heavysaturn commented:
did you forget to `git add` the tox.ini file?

hugovk commented:
No, there is no tox.ini file needed. 

This is for linting on the CI.

See https://github.com/OperationCode/resources_api/issues/16 for adding unit tests.

heavysaturn commented:
flake8 comes with `flake8 --install-hook git`, allowing you to add a flake8 pre-commit hook in a one-liner. This means you can lint your local stuff automatically before you commit it, That way you won't be forced to make additional commits just to fix linting errors discovered by the CI. This is a far better workflow than just leaving it to the CI.

If we're linting locally, stuff like `exclude=.venv` will be important. It's not unlikely that `max_line_length` should be changed and that certain linting codes should be ignored (B311 and S311 are notable examples, these will not allow you to do `import random`). 

For these reasons, I think a `tox.ini` would be a good inclusion

aaron-suarez commented on .travis.yml:
```diff
@@ -0,0 +1,11 @@
+script:
+  - flake8 app --statistics --count
+# TODO: implement pytests then delete this line and uncomment the next
+#  - pytest
```
Nit: When I initially committed this, I was missing the newline from the end of the file. I'd appreciate it if you'd please pick up my slack and stick that in 😉

hugovk replied:
Added :)

hugovk commented on app/__init__.py:
```diff
@@ -1,16 +1,17 @@
+    assert version_info >= (3, 7, 0)
+except AssertionError:
+    print('Warning Current Python version not supported')
+    print(f'Current Python version: {version_info}')
```
Reverted!

hugovk commented:
> 1. Install instructions for someone who is new to coding to do this linting

Added to Development Notes in the README.

> 2. Incorporate changes Leon suggested (in some form).

I'm not so sure how best to do tox.ini and the pre-commit hook. Could someone else take that up in a separate PR?

I've added a `.flake8` config file. How does that look?

aaron-suarez commented on README.md:
````diff
@@ -45,3 +45,10 @@ If you make changes to the models.py or other schemas, you need to run a migrati
+Before committing, please lint your code:
+
+```sh
+pip install --upgrade flake8
````
If flake8 is installed via virtualenv, is upgrading it necessary? Why do we need to run this command before running `flake8 app`?

aaron-suarez replied:
And to clarify, I'm legitimately asking because I don't know, not because I think we should remove this line. I'm sure it's fine, I just want to know why it's necessary.

hugovk replied:
It's not necessary to run it right before `flake8`, as long as it's been installed somehow first.

Some people might not install via virtualenv (I haven't for this PR). And it might be fine without the upgrade, but I generally include it in instructions to ensure the tools are kept up-to-date.

We could remove the upgrade if you prefer.

aaron-suarez replied:
No, I like it as is. I just assumed that anyone who gets to the bottom of the README would have followed the instructions and used virtualenv. But if they're knowledgable enough to know what kind of difference it makes, then they can adjust their command accordingly.

This seems like a good default for those who are kind of blindly following instructions and cargo culting trying to figure this stuff out.

heavysaturn commented:
The pre-commit hook comes with flake8, and you don't need to do anything to have that feature. You could perhaps mention in the documentation that `flake8 --install-hook git` will magically make flake8 run before any commit you try to make, and that if flake8 fails, it will stop the commit. This way they will not be allowed to commit code that does not pass the linter. (It is possible to bypass by doing `git commit --no-verify`, though)

As for the `tox.ini`, the `.flake8` is the same thing, so it's fine that you've done it with that filename instead. 

Here's the default config I usually have in my projects

```
[flake8]
max-line-length=120
exclude=.venv
ignore=B311,W503,E226,S311
import-order-style=pycharm
```

This assumes that the following flake8 plugins are installed:
- flake8-bugbear
- flake8-string-format
- flake8-todo
- flake8-import-order
- flake8-bandit

I will need @aaron-suarez or @wimo7083 to chime in on which of these we want for this project, and if you include them in the `requirements.txt` file you will also need to install them and do another round of linting and error correction, because they will probably discover some errors you haven't addressed yet.

Alternatively, I'd be happy to do all of the above in a separate PR, in which case this can just be merged because it looks pretty good.

aaron-suarez commented:
I like the PR as is, I would like to get it merged so we can rebase all of our existing PRs and move on. I would like to incorporate flake8-bandit for sure, we can look at the other ones. Since @heavysaturn is willing to do that work in a different PR and appears to be more familiar with these types of configuration files, he can do that PR.

@wimo7083, you had requested changes, and as far as I can tell they are incorporated. Can we go ahead and merge this?

aaron-suarez commented:
Probably need to rebase to fix the conflicts from merging #26... sorry about that
